tracing-opentelemetry/src/subscriber.rs

```diff
@@ -85,18 +87,22 @@ impl WithContext {
 }
 
 fn str_to_span_kind(s: &str) -> Option<otel::SpanKind> {
-    if s.eq_ignore_ascii_case("SERVER") {
-        Some(otel::SpanKind::Server)
-    } else if s.eq_ignore_ascii_case("CLIENT") {
-        Some(otel::SpanKind::Client)
-    } else if s.eq_ignore_ascii_case("PRODUCER") {
-        Some(otel::SpanKind::Producer)
-    } else if s.eq_ignore_ascii_case("CONSUMER") {
-        Some(otel::SpanKind::Consumer)
-    } else if s.eq_ignore_ascii_case("INTERNAL") {
-        Some(otel::SpanKind::Internal)
-    } else {
-        None
+    match s.to_ascii_lowercase().as_str() {
+        "server" => Some(otel::SpanKind::Server),
+        "client" => Some(otel::SpanKind::Client),
+        "producer" => Some(otel::SpanKind::Producer),
+        "consumer" => Some(otel::SpanKind::Consumer),
+        "internal" => Some(otel::SpanKind::Internal),
+        _ => None,
+    }
+}
+
+fn str_to_status_code(s: &str) -> Option<otel::StatusCode> {
+    match s.to_ascii_lowercase().as_str() {
+        "unset" => Some(otel::StatusCode::Unset),
+        "ok" => Some(otel::StatusCode::Ok),
+        "error" => Some(otel::StatusCode::Error),
+        _ => None,
```

Comment:
similarly, i think we should prefer `eq_ignore_ascii_case` for performance reasons:
```suggestion
    match s {
        s if s.eq_ignore_ascii_case("unset") => Some(otel::StatusCode::Unset),
        s if s.eq_ignore_ascii_case("ok") => Some(otel::StatusCode::Ok),
        s if s.eq_ignore_ascii_case("error") => Some(otel::StatusCode::Error),
        _ => None,
```